main/src/ca/uwaterloo/flix/api/Flix.scala

```diff
@@ -678,4 +674,59 @@ class Flix {
     threadPool.shutdown()
   }
 
+  /**
+    * Extends the set of known Java classes and interfaces with those in the given JAR-file `p`.
+    */
+  private def extendKnownJavaClassesAndInterfaces(p: Path): Unit = {
+    knownClassesAndInterfaces = knownClassesAndInterfaces ++ getPackageContent(getClassesAndInterfacesOfJar(p))
+  }
+
+  /**
+    * Returns all Java classes and interfaces in the current Java Platform.
+    */
+  private def getJavaPlatformClassesAndInterfaces(): MultiMap[List[String], String] = {
+    getPackageContent(ClassList.TheList)
+  }
+
+  /**
+    * Returns the names of all classes and interfaces in the given JAR-file `p`.
+    */
+  private def getClassesAndInterfacesOfJar(p: Path): List[String] = {
+    Using(new ZipFile(p.toFile)) { zip =>
+      val result = mutable.ListBuffer.empty[String]
+      val iterator = zip.entries()
+      while (iterator.hasMoreElements) {
+        val entry = iterator.nextElement()
+        val name = entry.getName
+        if (name.endsWith(".class")) {
+          result += name
+        }
+      }
+      result.toList
+    }.get
+  }
+
+  /**
+    * Returns a multimap from Java packages to sub-packages, classes, and interfaces.
+    */
+  private def getPackageContent(l: List[String]): MultiMap[List[String], String] = {
+    l.foldLeft[MultiMap[List[String], String]](MultiMap.empty) {
+      case (acc, clazz) =>
+        // Given a string `java/util/zip/ZipUtils.class` we convert it to the list `java :: util :: zip :: ZipUtils`.
+        val clazzPath = clazz.stripSuffix(".class").split('/').toList
```

Comment:
Does it work on Windows with the backslash?

Reply:
Its actually a zip file, so its /